> Fast test — Cannot build ClickHouse

For some reason compiler reports unreachable code for branch under `if constexpr`:
```
2022-12-01 07:34:58 /ClickHouse/src/Storages/MergeTree/MergeTreeIndexGranuleBloomFilter.cpp:102:31: error: code will never be executed [-Werror,-Wunreachable-code]
2022-12-01 07:34:58             read_size = filter->getFilter().size() * sizeof(BloomFilter::UnderType);
2022-12-01 07:34:58                               ^~
2022-12-01 07:34:58 /ClickHouse/src/Storages/MergeTree/MergeTreeIndexGranuleBloomFilter.cpp:119:38: error: code will never be executed [-Werror,-Wunreachable-code]
2022-12-01 07:34:58             write_size = bloom_filter->getFilter().size() * sizeof(BloomFilter::UnderType);
```

Probably related to https://github.com/llvm/llvm-project/issues/57123 ?

--- a/src/Storages/MergeTree/MergeTreeIndexGranuleBloomFilter.cpp
+++ b/src/Storages/MergeTree/MergeTreeIndexGranuleBloomFilter.cpp
@@ -98,8 +98,9 @@ void MergeTreeIndexGranuleBloomFilter::deserializeBinary(ReadBuffer & istr, Merg
     for (auto & filter : bloom_filters)
     {
         filter = std::make_shared<BloomFilter>(bytes_size, hash_functions, 0);
-        if constexpr (std::endian::native == std::endian::big)
-            read_size = filter->getFilter().size() * sizeof(BloomFilter::UnderType);
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+        read_size = filter->getFilter().size() * sizeof(BloomFilter::UnderType);
+#endif
         istr.readStrict(reinterpret_cast<char *>(filter->getFilter().data()), read_size);
     }
 }
@@ -115,8 +116,9 @@ void MergeTreeIndexGranuleBloomFilter::serializeBinary(WriteBuffer & ostr) const
     size_t write_size = (bits_per_row * total_rows + atom_size - 1) / atom_size;
     for (const auto & bloom_filter : bloom_filters)
     {
-        if constexpr (std::endian::native == std::endian::big)
-            write_size = bloom_filter->getFilter().size() * sizeof(BloomFilter::UnderType);
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+        write_size = bloom_filter->getFilter().size() * sizeof(BloomFilter::UnderType);
+#endif
         ostr.write(reinterpret_cast<const char *>(bloom_filter->getFilter().data()), write_size);
     }
 }

